### PR TITLE
core: avoid webinspector global pollution

### LINKED ISSUE #2545: Gulp lighthouse import conflicts with gulp-autoprefixer
I have encountered a strange issue and I am unsure on how to investigate this further. When both `gulp-autoprefixer` and `lighthouse` are imported into gulp (even, if in two separate files); `autoprefixer` does not run, nor does it return any errors.

Steps to replicate:
1. `npm install`
2. Setup a gulp task which executes `gulp-autoprefixer`
```js
gulp.task('styles', function() {
  return gulp.src('./style.css')
    .pipe(autoprefixer())
    .pipe(gulp.dest('./public'));
});
```
3. Running `gulp styles` in the command-line should run the above task and output the file style.css in the folder `public`
4. Import `lighthouse`
```js
const lighthouse = require('lighthouse');
```
5. Re-running `gulp styles` will no longer output any files.

I have created a fork to demonstrate further: https://github.com/IncredibleWeb/lighthouse/tree/feature/gulp_recipe_conflicts_with_autoprefixer/docs/recipes/gulp

### PR DESCRIPTION
**Pull request by castilloandres**
**Summary**

**What kind of change does this PR introduce?**
Allow portions of the DevTools frontend (Web Inspector) to be used in lighthouse by running them in a separate context.

**Is this a bugfix, feature, refactoring, build related change, etc?**
Refactoring.

**Describe the need for this change**
Avoid polluting the global scope.

fixes #2545

### CODE REVIEW

File: lighthouse-core/lib/web-inspector.js [hunk 1]
```diff
@@ -10,7 +10,11 @@
  * Stubbery to allow portions of the DevTools frontend to be used in lighthouse. `WebInspector`
  * technically lives on the global object but should be accessed through a normal `require` call.
  */
-module.exports = (function() {
+
+const vm = require('vm');
+const fs = require('fs');
+
+const getWebInspector = () => {
```
[the remark this answers is not in the document]
Comment (castilloandres) [reply lighthouse-core/lib/web-inspector.js]: @paulirish yes! Will definitely look into it! Thanks for the feedback 👍

**Comment by patrickhulce** [comment]
Thanks very much for tackling this beast in our codebase @castilloandres!! One other route I'll throw out there in case you're interested and the VM problems prove to be too thorny.

We only have 2 more uses of WebInspector at all left in our codebase: Font size and incremental JSON parsing (font-size.js, trace-parser.js). It'd be a huge win if we could eliminate it, so any work there would be a double victory for global pollution and removing a dependency :)

**Comment by castilloandres** [comment]
I isolated the WebInspector (cc @brendankenny) but I am very interested in re-implementing the `font-size` gatherer just need a bit more time. In the mean time maybe we can proceed with this approach?

**Comment by brendankenny** [comment]
@castilloandres this has been awesome work, but it looks like the font-size approach is going to eliminate webinspector pollution once and for all. So at least that's the positive side of things :)

Closing in favor of #6209

**Pull request closed without merging**